src/main/frontend/components/select.cljs

```diff
@@ -0,0 +1,77 @@
+(ns frontend.components.select
+  "Generic component for fuzzy searching items to select an item. See
+  select-config to add a new use or select-type for this component. To use the
+  new select-type, set :ui/open-select to the select-type. See
+  :select-graph/open command for an example."
+  (:require [frontend.modules.shortcut.core :as shortcut]
+            [frontend.context.i18n :as i18n]
+            ;; Would prefer to depend on handler.repo but unable to because
+            ;; of a circular dependency
+            [frontend.components.repo :as repo]
```

Comment:
The exact circular dependency was `frontend.handler.repo -> frontend.modules.shortcut.core -> frontend.modules.shortcut.data-helper -> frontend.modules.shortcut.config -> frontend.components.commit -> frontend.handler.repo`. I didn't try to take this on because I'm not too familiar with code organization just yet and I did see a related TODO for repo/open-repo-url

Reply:
I just addressed this in e0ab4f9599b09c82ac8b1a9add306ba008c25008.

Reply:
Thanks. With the :graph/switch event, is it ok to call directly without the safety check in https://github.com/logseq/logseq/blob/03f30a9391ff7eafd8a3ef94771b12ee6955770a/src/main/frontend/components/repo.cljs#L31? If it's not ok, I could see someone misusing this event and we should probably move the safety check into the event

Reply:
It's probably not ok 😄  I think we can delete the fn `switch-repo-if-writes-finished?` here and move its body to the :graph/switch event.